### ISSUE TITLE
Purging the shared cache does not work in debug mode

### ISSUE THREAD

**Issue by Mynyx** [contao/contao#4193]
### Affected version(s)

4.13.0

### Description

The function "Purge the shared cache" in back end maintenance does not work in debug mode.

Apparently the issue is related to the different environment. But although there exists no HTTP cache in the dev environment, a successful entry is added to the system log. Therefore, it seems reasonable to always clear the prod HTTP cache.

**Comment by fritzmg** [comment]
What exactly does not work?

**Comment by Mynyx** [comment]
The cached versions of the front end pages are not removed.

**Comment by fritzmg** [comment]
I can confirm the problem. Reproduction steps:

1. Enable the shared cache in the site structure.
2. Create a request to a Contao page and make sure it responds with `Contao-Cache: fresh` or `Contao-Cache: miss/store`.
3. Enable the debug mode.
4. Go into the maintenance section and purge the HTTP cache.
5. Make another request to the same URL.

The response will still say `Contao-Cache: fresh`.

**Comment by aschempp** [comment]
Well it _does_ work, but it does not do what you expect. The cache is always cleared in the current environment, and clearing the cache in `dev` does not clear the cache in `prod`. Since we (correctly) use the Cache proxy to clear the cache, I can only think of two options:
1. disable/remove the option in `dev`
2. add a description that explicitly states it clears the current environment

But option 2 does not make much sense since `dev` never has a cache (at least not in the Contao Managed Edition) 😅

**Comment by aschempp** [comment]
Also pretty sure this happens in 4.9 as well, maybe @Mynyx you can test that?

**Comment by fritzmg** [comment]
We could may be clear it for the prod environment (or both environments) in a separate process instead.

**Comment by aschempp** [comment]
A separate process? You mean run a command?

**Comment by aschempp** [comment]
ping @fritzmg

**Comment by fritzmg** [comment]
> A separate process? You mean run a command?

Yes, `contao:automator purgePageCache` for example (or a dedicated command).

**Comment by michb** [comment]
I think this issue is also affecting the automatic http-cache invalidation. I've currently tested it on **Contao 5.7.3**.

In my case, when Debug Mode is enabled, any data changes made in the backend (e.g., in `tl_content`) only seem to trigger invalidation for the `dev` cache. Thats totally understandable. The `prod` cache remains untouched and continues to serve stale data.
Since the underlying database changes affect both environments, it would make sense to always invalidate the `prod` HTTP cache as well. Otherwise, an inconsistent state where the production frontend doesn't reflect the backend changes.

Steps to Reproduce
1. Enable HTTP Cache in the site settings.
2. Visit a page in production mode to fill the `prod` cache, check for `Contao-Cache` header
3. Enable Debug Mode in the Contao Backend.
4. Modify a content element in tl_content (e.g., change a headline).
5. Verify `dev` cache: Reload the page in Debug Mode. The changes are visible (cache was correctly invalidated or bypassed).
6. Verify `prod` cache: Disable Debug Mode (or open the page in an incognito window/different browser). The page still shows the old content because the `prod` cache was not invalidated.

**Comment by leofeyer** [comment]
As discussed in the call, @aschempp will look into this and give it a try.
